backend/apps/ecommerce/vipps_utils.py

```diff
@@ -189,30 +192,23 @@ def _refresh_access_token(self) -> Tuple[str, str]:
         expires_on = timezone.make_aware(datetime.datetime.fromtimestamp(int(token_response["expires_on"])))
         return access_token, expires_on
 
-    def _fetch_access_token(self) -> None:
-        # Get Vipps access token from db or fetch new if necessary
-        tokens = VippsAccessToken.objects
-        if not tokens.filter(expires_on__gte=timezone.now()).exists():
-            # No valid token in database, delete all stale tokens and get new token
-            if tokens.exists():
-                tokens.all().delete()
-            new_token = VippsAccessToken()
-            self._access_token, expires_on = self._refresh_access_token()
-            new_token.token = self._access_token
-            new_token.expires_on = expires_on
-            new_token.save()
-
-        self._access_token = tokens.filter(expires_on__gte=timezone.now()).first().token
-
     @property
     def access_token(self) -> str:
         """Checks if access token already obtained
         Returns:
             str: Access Token
         """
-        if self._access_token is None:
-            self._fetch_access_token()
-        return self._access_token
+        valid_token: Optional[VippsAccessToken] = VippsAccessToken.objects.filter(
+            expires_on__gte=timezone.now() + timezone.timedelta(minutes=5)
```

Comment:
Check for one slightly in the future to avoid edge cases where it's just about to expire